src/rhsmlib/facts/host_collector.py

```diff
@@ -68,8 +71,16 @@ def get_all(self):
 
         locale_info = {}
         effective_locale = 'Unknown'
-        if locale.getdefaultlocale()[0] is not None:
-            effective_locale = ".".join(filter(None, locale.getdefaultlocale()))
+        # When there is no locale set (system variable LANG is unset),
+        # then this is value returned by locale.getdefaultlocale()
+        # Tuple contains: (language[_territory], encoding identifier)
+        default_locale = (None, None)
+        try:
+            default_locale = locale.getdefaultlocale()
+        except ValueError as err:
+            log.warning("Unable to get default locale (wrong system variable LANG): %s" % err)
```

Comment:
May be, we can try here to get system variable LANG and use it, but I prefer to set `effective_locale` to "Unknown", when it doesn't make any sense to other software too and it has no effect (translation using gettext, etc.).